shared/editor/nodes/ToggleBlock.ts

```diff
@@ -0,0 +1,605 @@
+import concat from "lodash/concat";
+import filter from "lodash/filter";
+import flatten from "lodash/flatten";
+import forEach from "lodash/forEach";
+import isEmpty from "lodash/isEmpty";
+import isNil from "lodash/isNil";
+import map from "lodash/map";
+import some from "lodash/some";
+import {
+  chainCommands,
+  createParagraphNear,
+  joinForward,
+  joinTextblockBackward,
+  newlineInCode,
+  splitBlock,
+  wrapIn,
+} from "prosemirror-commands";
+import { ParseSpec } from "prosemirror-markdown";
+import { NodeSpec, NodeType, Node as ProsemirrorNode } from "prosemirror-model";
+import { Command, Plugin, PluginKey, TextSelection } from "prosemirror-state";
+import {
+  Decoration,
+  DecorationSet,
+  DecorationSource,
+  EditorView,
+  NodeView,
+} from "prosemirror-view";
+import { v4 } from "uuid";
+import Storage from "../../utils/Storage";
+import {
+  createParagraphBefore,
+  split,
+  lift,
+  liftNext,
+  sinkBlockInto,
+  liftLastBlockOutOf,
+  ancestors,
+  suchThat,
+  depth,
+  furthest,
+} from "../commands/toggleBlock";
+import { chainTransactions } from "../lib/chainTransactions";
+import { MarkdownSerializerState } from "../lib/markdown/serializer";
+import { findBlockNodes } from "../queries/findChildren";
+import toggleBlocksRule from "../rules/toggleBlocks";
+import Node from "./Node";
+
+enum Action {
+  INIT,
+  CHANGE,
+  FOLD,
+  UNFOLD,
+}
+
+export default class ToggleBlock extends Node {
+  static pluginKey = new PluginKey<DecorationSet>("toggleBlockPlugin");
+
+  get name() {
+    return "container_toggle_block";
```

Comment:
```suggestion
    return "container_toggle";
```

Reply:
Just pushed this change. Make sure to _erase_ all toggle blocks in your existing test docs before pulling latest changes @tommoor

Reply:
roger that